tutorials/source-modeling/plot_beamformer_lcmv.py

```diff
@@ -0,0 +1,169 @@
+The beamforming method applied in this tutorial is the linearly constrained
+minimum variance (LCMV) beamformer [1]_ which operates on time series.
+Frequency-resolved data can be reconstructed with the dynamic imaging of
+coherent sources (DICS) beamforming method [2]_.
+"""
+# Author: Britta Westner <britta.wstnr@gmail.com>
+#
+# License: BSD (3-clause)
+
+# sphinx_gallery_thumbnail_number = 5
+
+import mne
+from mne.datasets import sample
+from mne.beamformer import make_lcmv, apply_lcmv
+
+###############################################################################
+# Data processing
+# ---------------
+# We will use the sample data set for this tutorial and reconstruct source
+# activity on the trials with left auditory stimulation. Note that
+# beamformers are usually computed in a :class:`volume source space
+# <mne.VolSourceEsitmate>`, as a visualization
+# of only the surface activation can misrepresent the data.
+
+data_path = sample.data_path()
+subjects_dir = data_path + '/subjects'
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+fwd_fname = data_path + '/MEG/sample/sample_audvis-meg-vol-7-fwd.fif'
+
+# Read forward model
+forward = mne.read_forward_solution(fwd_fname)
+
+# Read the raw data
+raw = mne.io.read_raw_fif(raw_fname, preload=True)
+raw.info['bads'] = ['MEG 2443']  # bad MEG channel
+
+# Set up the epoching
+event_id = 1  # those are the trials with left-ear auditory stimuli
+tmin, tmax = -0.2, 0.5
+events = mne.find_events(raw)
+
+# pick relevant channels
+raw.pick(['meg', 'eog'])  # pick channels of interest
+
+# Create epochs
+proj = False  # already applied
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax,
+                    baseline=(None, 0), preload=True, proj=proj,
+                    reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+
+# Visualize averaged sensor space data
+evoked = epochs.average()
+evoked.plot_joint()
+
+###############################################################################
+# Computation of covariance matrices
+# ----------------------------------
+# The spatial filter is computed from two ingredients: the forward model
+# solution that we read from disk above and the covariance matrix of the data.
+# The data covariance matrix will be `inverted`_ during the spatial filter
+# computation, so it is valuable to plot the covariance matrix and its
+# eigenvalues to gauge whether matrix inversion will be possible.
+# Also, because we want to combine different channel types (magnetometers and
+# gradiometers), we need to account for the different amplitude scales of these
+# channel types. To do this we will supply a noise covariance matrix to the
+# beamformer, which will be used for whitening.
+
+data_cov = mne.compute_covariance(epochs, tmin=0.05, tmax=0.25,
+                                  method='empirical')
+noise_cov = mne.compute_covariance(epochs, tmin=tmin, tmax=0,
+                                   method='empirical')
```

Comment:
Here (or in the paragraph above) it would be good to spell out why these particular tmin/tmax values are chosen.  Is there a reason *not* to choose data tmin as 0?  It looks like peak response ends around 0.15, so why is tmax 0.25 here? would 0.4 be too much? etc.

Reply:
Done!